### PR TITLE
fix: set token network filter when adding network from dapp

### PR DESCRIPTION
**Pull request by sahar-fehri**
## **Description**

PR to setTokenNetworkFilter value when user is adding a new network from dapp

## **Related issues**

Fixes:

## **Manual testing steps**

1. Click on current network
2. Add new network from chainlist
3. Confirm switching to this network
4. Go back to wallet and you should see your new network with the correct token list

### **Before**

https://github.com/user-attachments/assets/03b43fd3-239e-47e5-a419-2c87a76e84d5

### **After**

https://github.com/user-attachments/assets/c11c4fb4-396f-4db9-b359-2c91108a0b42

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by vinnyhoward — DISMISSED** [review]

**Review by bergeron — DISMISSED** [review]

**Review by Daniel-Cross — DISMISSED** [review]
Looks good!

File: app/components/Approvals/SwitchChainApproval/SwitchChainApproval.test.tsx [hunk 1]
```diff
@@ -81,4 +95,29 @@ describe('SwitchChainApproval', () => {
...
+      'setTokenNetworkFilter',
+    );
+    mockApprovalRequest({
+      type: ApprovalTypes.SWITCH_ETHEREUM_CHAIN,
+      requestData: {
+        rpcUrl: URL_MOCK,
+      },
+      // TODO: Replace "any" with type
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } as any);
```
Comment (OGPoyraz) [inline app/components/Approvals/SwitchChainApproval/SwitchChainApproval.test.tsx:112]:
```suggestion
     } as ApprovalRequest<{
      rpcUrl: string;
    }>);
```

File: app/components/Approvals/SwitchChainApproval/SwitchChainApproval.tsx [hunk 2]
```diff
@@ -15,17 +19,40 @@ const SwitchChainApproval = () => {
...
+        PreferencesController.setTokenNetworkFilter({
+          [approvalRequest?.requestData?.chainId]: true,
+        });
+      } else {
+        PreferencesController.setTokenNetworkFilter({
+          ...tokenNetworkFilter,
+          [approvalRequest?.requestData?.chainId]: true,
+        });
+      }
+    }
```
Comment (OGPoyraz) [inline app/components/Approvals/SwitchChainApproval/SwitchChainApproval.tsx:41]:
```suggestion
if (isPortfolioViewEnabled()) {
    const { PreferencesController } = Engine.context;
    PreferencesController.setTokenNetworkFilter({
      ...(isAllNetworks ? tokenNetworkFilter : {}),
      [approvalRequest?.requestData?.chainId]: true,
    });
}
```

**Review by OGPoyraz — DISMISSED** [review]

**Review by Daniel-Cross — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Review by salimtb — APPROVED** [review]

**Comment by salimtb** [comment]
tested and approved ✅

**Pull request merged**